### ISSUE TITLE
[QUESTION]: Should we use torch.inference_mode() for trainer.test()

### ISSUE THREAD

**Issue by KnathanM** [chemprop/chemprop#682]
A while back we decided to wrap `trainer.predict` with `torch.inference_mode()` https://github.com/chemprop/chemprop/pull/469#discussion_r1346434635

Is that something we should do with `trainer.test` in `train.py` as well?

**Comment by JacksonBurns** [comment]
lightning automatically sets the model to `eval` mode and disables gradient calculation as per the docs, see the second note for `test_step` at this link: https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.core.LightningModule.html#lightning.pytorch.core.LightningModule.test_step

the docs also imply that you should not need to wrap predict either, see https://lightning.ai/docs/pytorch/stable/deploy/production_basic.html#predict-step-with-your-lightningmodule

**Comment by kevingreenman** [comment]
I think our conclusion in #469 was that it probably wasn't necessary but wouldn't hurt, so we included it. Perhaps we should include it for consistency, but I don't think it matters either way.

**Comment by KnathanM** [comment]
Using `with torch.inference_mode()` was a patch until lightning could fix some dependency issues. These have been fixed now and `inference_mode = True` is the default. I'll put in a PR to remove the  `with torch.inference_mode()` which will close this issue.

https://github.com/Lightning-AI/pytorch-lightning/issues/11018
https://github.com/Lightning-AI/pytorch-lightning/pull/15034
